### PR TITLE
fix(sync): consolidate Copilot + Members onto sync_events framework

### PR DESCRIPTION
**Pull request by studert**
## Summary

- **Copilot UI was frozen at 2026-03-25.** The three Copilot read paths (`getCopilotSyncStatus`, the activity timeline in `getCopilotSeatDetail`, and `getCopilotBillingSyncHistory`) still queried the legacy `github_sync_events` table — but writes had moved to the new `sync_events` framework on that date.
- **GitHub Members had the inverse problem.** `confirmGitHubSync` still wrote to `github_sync_events`, while the `/settings/sync` dashboard reads only `sync_events` — so the `github_members` row on the dashboard never lit up after a manual sync.
- **Both flows now route through `src/lib/sync/framework.ts`.** Copilot reads use new helpers in `src/lib/sync/queries.ts`; members writes wrap `withSyncLock({ sourceType: 'github_members' })` around the existing transaction.

## What changed

| File | Change |
|---|---|
| `src/lib/sync/queries.ts` (new) | `getLastCompletedSyncEvent`, `listSyncEvents`, `mapOutcomeToLegacyStatus`, `mapOutcomeForDisplay` — typed helpers shared by both Copilot and (future) members read surfaces. Orders by `started_at DESC` to ride the existing `sync_events_source_started_idx` index. |
| `src/actions/copilot.ts` | `getCopilotSyncStatus` now reads from `sync_events`. |
| `src/actions/copilot-data.ts` | `getCopilotSeatDetail`'s activity timeline and `getCopilotBillingSyncHistory` migrated to `listSyncEvents`. Status mapping uses the new `mapOutcomeForDisplay` helper (no `?? evt.outcome` fallback at call sites). |
| `src/actions/github-sync.ts` | `confirmGitHubSync` wrapped in `withSyncLock`. Members-specific counts mapped onto the generic create/update/skip/error shape. Both validation errors and pre-existing match conflicts flow into `errorCount` so the dashboard outcome is accurate. New `GitHubMemberSyncCounts` type exported. Action now returns `eventId` for future UI handles. Legacy `getSyncHistory` deleted (was dead — superseded by `getSyncHistory` in `actions/sync.ts`). |
| `src/actions/sync.ts` | `triggerSync` dispatcher returns `null` for `github_members` (interactive flow via the sheet; mirrors `anthropic_team_invoices` pattern). |
| `src/lib/copilot-sync.ts` | Dead `runCopilotSync` removed; unused imports cleaned up. |
| `src/lib/sync/sources/github-members.ts` | Deleted — the pass-through stub had zero callers. |
| `src/types/index.ts` | Orphaned `GitHubSyncEvent` / `NewGitHubSyncEvent` / `GitHubSyncStatus` re-exports removed. |
| `tests/unit/sync/queries.test.ts` | Covers all branches of `mapOutcomeToLegacyStatus`. |
| `tests/integration/sync/copilot-read-paths.test.ts` | Regression: seeds `sync_events` rows for both `github_copilot_billing` and `github_members` and asserts the helpers find them. Includes lock-respects-source-filter coverage. |

The legacy `github_sync_events` table itself is **kept in place** (no schema migration). After one release at green we can drop it in a follow-up PR.

## Out of scope

- Dropping the `github_sync_events` table (follow-up).
- Converging `listSyncEvents` with `getSyncHistory` in `actions/sync.ts` — they serve different concerns (manual/scheduled partitioning) and a real merge would change the public `/settings/sync` action signature.
- Multi-connection support for Copilot syncs — the new `sync_events` table is connection-agnostic; documented in the spec.

## Verification

End-to-end verified on Neon branch `wt/copilot-bugfix` (forked from `main`):

1. `/settings/sync` dashboard — `github_copilot_billing` row reflects the last cron-driven sync (about 7 hours ago), not 2026-03-25.
2. `/copilot/seats/1264796` "Sync History" panel — entries up to **2026-05-21** (today). Before this PR, this timeline was frozen since 2026-03-25.
3. Zero console errors / warnings across `/settings/integrations`, `/settings/sync`, `/copilot`, `/copilot/seats`, `/copilot/seats/:id`, `/copilot/billing`.
4. Screenshots: `specs/030-copilot-sync-read-migration/verification-{sync-dashboard,seat-sync-history}.png`.

## Test plan

- [x] `pnpm typecheck` — clean
- [x] `pnpm lint` — clean
- [x] `pnpm test` — 286/286 passing
- [x] `pnpm test:integration tests/integration/sync/copilot-read-paths.test.ts` — 5/5 passing on the branch DB
- [x] Manual smoke on a real Neon branch via the dev-only `/api/agent/session` mint
- [ ] (Reviewer) Trigger an interactive members sync via the sheet on preview, confirm the `github_members` row on `/settings/sync` advances
- [ ] (Reviewer) Wait for the next cron Copilot sync; confirm the Copilot row continues to advance

## Spec & implementation notes

- Detailed plan: [`specs/030-copilot-sync-read-migration/plan.html`](specs/030-copilot-sync-read-migration/plan.html)
- Running notes (design decisions, deviations, tradeoffs, open questions): [`specs/030-copilot-sync-read-migration/implementation-notes.html`](specs/030-copilot-sync-read-migration/implementation-notes.html)

Open question worth a look during review: `CopilotSyncSection` and the action-only `getCopilotBillingSyncHistory` no longer appear to be consumed by any rendered UI surface (they were removed in a prior refactor). The actions remain correct against the new framework, but a follow-up PR could delete them if the UI absence is intentional. Flagged in the notes; not addressed here.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/actions/github-sync.ts [hunk 1]
```diff
@@ -162,403 +170,410 @@ export async function confirmGitHubSync(
     return { success: false, error: fetchResult.error };
   }
```
[the remark this answers is not in the document]
Comment (studert) [reply src/actions/github-sync.ts]: Good catch — applied. Moved `fetchAndMatchMembers()` inside the `withSyncLock` callback so concurrent confirms now contend at the advisory lock before issuing any GitHub calls. The second confirm fails fast with "Sync already in progress" and the rate-limit budget is preserved.

Bonus: a fetch failure inside the lock now also surfaces in the sync history (the callback returns counts with `errorCount: 1` + the fetch error) — previously a rate-limit error on this path was invisible in the dashboard.

`fetchGitHubSyncPreview` (used by the sheet on open) intentionally still runs outside any lock, since it's a read-only preview that multiple admins should be able to inspect concurrently. Fixed in a2a7e6b.

File: src/actions/github-sync.ts [hunk 2]
```diff
@@ -162,403 +170,410 @@ export async function confirmGitHubSync(
...
+        // (duplicate email/login) and pre-existing match conflicts that
+        // blocked resolution. Both warrant a non-success outcome on the
+        // sync_events row so the dashboard surfaces them.
+        return {
+          createdCount: result.importedCount + result.createdCount,
+          updatedCount: result.enrichedCount + result.manuallyMatchedCount,
+          skippedCount: result.skippedCount,
+          errorCount: result.errors.length + result.conflictCount,
+          errorMessage:
+            result.errors.length > 0 ? result.errors.join("; ") : null,
```
[the remark this answers is not in the document]
Comment (studert) [reply src/actions/github-sync.ts]: Applied. Extracted a `summarizeErrors(errors[])` helper into `src/lib/sync/error-message.ts` with the same 1000-char cap that `withSyncLock` uses on its own failure path. The helper greedily includes complete messages until the budget runs out, then appends a `… (+N more)` suffix so the UI surfaces the dropped count rather than rendering a truncated mid-message string.

Covered by a 5-case unit test in `tests/unit/sync/error-message.test.ts`, including the edge case where the very first message alone overflows the cap. Fixed in a2a7e6b.

File: specs/030-copilot-sync-read-migration/plan.html [hunk 3]
```diff
@@ -0,0 +1,631 @@
...
+import type { SyncSourceType } from "@/lib/sync/framework";
+
+export async function getLastCompletedSyncEvent(sourceType: SyncSourceType) {
+  return db.query.syncEvents.findFirst({
+    where: and(
+      eq(syncEvents.sourceType, sourceType),
+      ne(syncEvents.outcome, "in_progress"),
+    ),
+    orderBy: [desc(syncEvents.completedAt)],
+  });
```
[the remark this answers is not in the document]
Comment (studert) [reply specs/030-copilot-sync-read-migration/plan.html:378]: Applied. Updated the `plan.html §8.0` snippet to match the merged code: ordering is `desc(syncEvents.startedAt)` (with a comment explaining the index motivation), `listSyncEvents` projection includes `skippedCount`, and the `mapOutcomeForDisplay` helper that was added post-review is now documented in the snippet too.

The implementation notes in `specs/030-copilot-sync-read-migration/implementation-notes.html` also already captured the `startedAt`-vs-`completedAt` decision (three reviewers independently flagged it during /simplify), so the spec is now consistent end-to-end. Fixed in a2a7e6b.
